gpbft/participant.go

```diff
@@ -48,7 +42,19 @@ type Participant struct {
 	// protocol round for which a strong quorum of COMMIT messages was observed,
 	// which may not be known to the participant.
 	terminatedDuringRound uint64
-	validator             *cachingValidator
+	// progression is the atomic reference to the current GPBFT instance being
+	// progressed by this Participant, or the next instance to be started if no such
+	// instance exists.
+	progression *atomicProgression
+	// messageCache maintains a cache of unique identifiers for valid messages or
+	// justifications that this Participant has received since the previous instance.
+	// It ensures that only relevant messages or justifications are retained by
+	// automatically pruning entries from older instances as the participant
+	// progresses to the next instance.
+	//
+	// See Participant.finishCurrentInstance, Participant.validator.
+	messageCache *caching.GroupedSet
```

Comment:
I ended up pulling this back into participant, so that on finish instance we can call it to evict older messages.

I think having this in Participant might come handy for implementing optimisations or equivocation protections that require cross-round or cross-instance state of received messages.